### PR TITLE
GET Blob Structured Message

### PR DESCRIPTION
**Pull request by jaschrep-msft**
GET Blob uses structured message when requesting a CRC on an unspecified range, or a range larger than 4 MiB.

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs [hunk 1]
```diff
@@ -1685,13 +1689,36 @@ private void SetNameFieldsIfNull()
...
+            bool? rangeGetContentMD5 = null;
+            bool? rangeGetContentCRC64 = null;
+            string structuredBodyType = null;
+            switch (validationOptions?.ChecksumAlgorithm.ResolveAuto())
+            {
+                case StorageChecksumAlgorithm.MD5:
+                    rangeGetContentMD5 = true;
+                    break;
+                case StorageChecksumAlgorithm.StorageCrc64:
+                    if (pageRange?.Length <= Constants.StructuredMessage.MaxDownloadCrcWithHeader)
```
Comment (jalauzon-msft) [inline sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs:1701]: Hmm, do we want to just always use Structured Message for this download? Is splitting the logic worth the efficiency gain of using the header? I guess the logic split is not very impactful in code, but I wonder if customers who are watching requests might be confused? Not really sure here, just thinking aloud.

File: sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs [hunk 2]
```diff
@@ -1725,8 +1753,15 @@ private void SetNameFieldsIfNull()
             long length = response.IsUnavailable() ? 0 : response.Headers.ContentLength ?? 0;
             ClientConfiguration.Pipeline.LogTrace($"Response: {response.GetRawResponse().Status}, ContentLength: {length}");
 
+            BlobDownloadStreamingResult result = response.ToBlobDownloadStreamingResult();
+            if (response.GetRawResponse().Headers.TryGetValue(Constants.StructuredMessage.CrcStructuredMessageHeader, out string _) &&
```
Comment (jalauzon-msft) [inline sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs:1757]: Do we want to fail if we asked for structured message but didn't get one back? Again, not sure here, just a thought.

**Review by jalauzon-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs (hunk 2 above)
Comment (jaschrep-msft) [reply sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs:1757]: Probably. Need to think on that more. Good catch.

File: sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs (hunk 1 above)
Comment (jaschrep-msft) [reply sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs:1701]: sticking with this for compatibility on certain error states.

**Pull request merged**